src/legacy/ui/public/agg_types/agg_param.d.ts

```diff
@@ -23,9 +23,13 @@ interface AggParam {
   type: string;
   name: string;
   required?: boolean;
+  dataTestSubj?: string;
   displayName?: string;
+  displayLabel?: string;
+  displayToolTip?: string;
   onChange?(agg: AggConfig): void;
   disabled?(agg: AggConfig): boolean;
+  shouldShow?(agg: AggConfig): boolean;
```

Comment:
We cannot simply rename that, since this is part of our API, and we have plugins that write custom AggConfigs and rely on the parameter `disabled` to continue working as it is.

I think we can just move the "disabled" state of the checkbox into a custom property on the `SwitchParamEditor` which will then just be set by the inline wrapper that I outlined in my other comment.